src/Passes/docs/src/DeveloperGuide/CodeQuality.md

```diff
@@ -0,0 +1,167 @@
+# Code quality and continuous integration
+
+## Using the manage tool
+
+Before making a pull request with changes to this library, please ensure that style checks passes, that the code compiles,
+unit test passes and that there are no erros found by the static analyser.
+
+To setup the CI environment, run following commands
+
+```sh
+source develop.env
+virtualenv develop__venv
+source develop__venv/bin/activate
+pip install -r requirements.txt
+```
+
+These adds the necessary environment variables to ensure that you have the `TasksCI` package and all required dependencies.
+
+To check the style, run
+
+```sh
+./manage stylecheck
+```
+
+To test that the code compiles and tests passes run
+
+```sh
+./manage test
+```
+
+Finally, to analyse the code, run
+
+```sh
+./manage lint
+```
+
+You can run all processes by running:
+
+```sh
+./manage runci
+```
+
+As `clang-tidy` and `clang-format` acts slightly different from version to version and on different platforms, it is recommended
+that you use a docker image to perform these steps. TODO(TFR): The docker image is not added yet and this will be documented in the future.
+
+## Running tests
+
+In order to run the tests, you first need to build the library. Assuming that this is already done and the corresponding build is in `Debug/`, run the tests from the `Debug` folder:
+
+```
+lit tests/ -v
+-- Testing: 2 tests, 2 workers --
+PASS: Quantum-Passes :: QirAllocationAnalysis/case1.ll (1 of 2)
+PASS: Quantum-Passes :: QirAllocationAnalysis/case2.ll (2 of 2)
+
+Testing Time: 0.27s
+  Passed: 2
+```
+
+The C++ test suite can also be ran from the debug by first building all targets:
+
+```sh
+cmake ..
+make
+```
+
+and then running following command:
+
+```sh
+ctest
+```
+
+## Writing C++ tests
+
+TODO(tfr): Write this section
+
+## Modifying code quality requirements
+
+This component is the largest part of this PR. The continuous integration component includes:
```

Comment:
> is the largest part of **this PR**

Consider making the documentation non-**PR-oriented**. After the PR is merged, it will be confusing for the reader which exactly PR is meant.

Also applicable to `IntroductionPass.md`, fragments "An important part of this PR", "developed in this PR".

Reply:
Corrected.